relay-server/src/actors/project_cache.rs

```diff
@@ -446,6 +447,35 @@ impl Services {
     }
 }
 
+/// Tracks how many projects are accessed every second.
+#[derive(Debug)]
+struct AccessTracker {
+    keys: HashSet<ProjectKey>,
+    last_reset: Instant,
+}
+
+impl AccessTracker {
+    fn new() -> Self {
+        Self {
+            keys: Default::default(),
+            last_reset: Instant::now(),
+        }
+    }
+
+    fn track(&mut self, project_key: ProjectKey) {
+        let now = Instant::now();
+        if (now - self.last_reset) > Duration::from_secs(1) {
```

Comment:
should we use [`elapsed`](https://doc.rust-lang.org/stable/std/time/struct.Instant.html#method.elapsed) instead here, which  returns duration you could compare?